### DISCUSSION TITLE
get/insert current safari URL+title

### DISCUSSION THREAD

**Discussion by zeltak** [SilentVoid13/Templater#314] [Help] [answered]
Hi all!

just discovered Templater and im in love ❤️  ;-)

i am wondering if anyone has a sniper/code to share to allow inserting the current safari URL+title (preferably in MD format)?

i apologize in advance as im not super tech literate but would love if the community can share some examples to help me start going!

thx so much

Z

**Comment by welpdx** [comment, marked as answer]
Could you please elaborate on your use case / workflow?

Are you copying the url from safari and wanting to paste it like `[text](copiedUrl)` into obsidian? (If so, try [this](https://gist.github.com/welpdx/7036f8121b59ee660201ea2c753d51f1).)
Or are you looking to create a new note or something?

**Comment by zeltak** [comment]
thx @welpdx !

> Are you copying the url from safari and wanting to paste it like [text](copiedUrl) into obsidian
yes exactly!

thx for the link that's cool. was wondering if there was any java script that could programatically take the URL from the current tab in safari without having to copy the link manually

best

Z

**Reply by welpdx** [reply]
Hi Z, 

I'm glad I was able to help!

Unfortunately, the Templater plugin allows you to create scripts and macros that work within the obsidian program. You can't really use Templater to do things outside of it. (Unless you like create a custom js function that calls other scripts which does you want. This is beyond me lol)

Since you are on a mac, you may be interested to look into [Keyboard Maestro](https://www.keyboardmaestro.com/main/), an automation/macro software. You can program it such that when you enter a hotkey it will grab the url from the current safari tab and edit the copied text so that when you paste it in Obsidian, it is in md. 

I'm not familiar with Maestro or with mac, so I'm sorry I can't help you there. But here are some posts related to your request on Maestro's forums and wiki. 
- [Get Current Window URL from Chrome or Safari](https://forum.keyboardmaestro.com/t/get-current-window-url-from-chrome-or-safari/7518)
- [Copy and paste webpage URL and title](https://forum.keyboardmaestro.com/t/copy-and-paste-webpage-url-and-title/16012)
- [Clipboards in Maestro Wiki](https://wiki.keyboardmaestro.com/Clipboards)

Also, there are Obsidian users who is on mac and is using Keyboard Maestro to automate Obsidian. Check [this](https://forum.obsidian.md/t/using-and-automating-obsidian-on-macos-meta-post/3129) out. You may find some useful scripts for Maestro there (similar to finding useful Templater scripts for Templater :D). 

Hope that helps.

**Reply by zeltak** [reply]
thx again that's fantastic!!!
